web/js/modules/layers/util.js

```diff
@@ -1288,3 +1295,59 @@ export function adjustStartDates(layers) {
 
   return Object.values(layers).forEach(applyDateAdjustment);
 }
+
+/**
+ * Change end dates for future layers to add 'futureTime' (ex: '3D')
+ * Adjust def.endDate and, if applicable, the last endDate in dateRange object
+ *
+ * @method adjustEndDates
+ * @param  {Array} layers array
+ * @returns {Array} array of layers
+ */
+export function adjustEndDates(layers) {
+  const applyDateAdjustment = (layer) => {
+    const { futureTime, dateRanges } = layer;
+    if (!futureTime) {
+      return;
+    }
+
+    const futureEndDate = getFutureLayerEndDate(layer);
+    layer.endDate = util.toISOStringSeconds(futureEndDate);
+
+    if (dateRanges.length) {
+      const lastDateRange = dateRanges[dateRanges.length - 1];
+      lastDateRange.endDate = util.toISOStringSeconds(futureEndDate);
+    }
+  };
+
+  return Object.values(layers).forEach(applyDateAdjustment);
+}
+
+/**
+ * Build end date for future layer
+ *
+ * @method mockFutureTimeLayerOptions
+ * @param  {Array} layers array
+ * @param  {String} mockFutureLayerParameters 'targetLayerId, mockFutureTime'
+ * @returns {Array} array of layers
+ */
+export function mockFutureTimeLayerOptions(layers, mockFutureLayerParameters) {
```

Comment:
... that this `mockFutureTimeLayerOptions` only has to add the `futureTime` property to the layer specified in the URL params and then `adjustEndDates` will do the rest, allowing this function to be simplified to:

```
export function mockFutureTimeLayerOptions(layers, mockFutureLayerParameters) {
  const urlParameters = mockFutureLayerParameters.split(',');
  const [targetLayerId, mockFutureTime] = urlParameters;

  if (targetLayerId && mockFutureTime) {
    layers[targetLayerId].futureTime = mockFutureTime;
  }
}
```

That way you don't have to repeat the logic of of setting the new endDate, or loop through all the layers again.